cpp/ql/src/Critical/OverflowDestination.qhelp

```diff
@@ -3,27 +3,31 @@
   "qhelp.dtd">
 <qhelp>
 <overview>
-<p>The bounded copy functions <code>memcpy</code>, <code>memmove</code>, <code>strncpy</code>, <code>strncat</code> accept a size argument. You should call these functions with a size argument that is derived from the size of the destination buffer. Using a size argument that is derived from the source buffer may cause a buffer overflow. Buffer overflows can lead to anything from a segmentation fault to a security vulnerability.</p>
-
-
+<p>The bounded copy functions <code>memcpy</code>, <code>memmove</code>, <code>strncpy</code>, <code>strncat</code> accept a size argument. 
+You should call these functions with a size argument that is derived from the size of the destination buffer. 
+Using a size argument that is derived from the source buffer may cause a buffer overflow. 
+Buffer overflows can lead to anything from a segmentation fault to a security vulnerability.</p>
 </overview>
+
 <recommendation>
-<p>Check the highlighted function calls carefully, and ensure that the size parameter is derived from the size of the destination buffer,
+<p>Check the highlighted function calls carefully. 
+Ensure that the size parameter is derived from the size of the destination buffer, and 
 not the source buffer.</p>
 
 <include src="aliasAnalysisWarning.qhelp" />
-
 </recommendation>
-<example><sample src="OverflowDestination.cpp" />
 
 
+<example>
+<p>
+The code below shows an example where <code>strncpy</code> is called incorrectly, without checking the size of the destination buffer.
+In the second example the call has been updated to include the size of the destination buffer.</p>
 
+<sample src="OverflowDestination.cpp" />
 </example>
-<references>
 
-<li><a href="http://cwe.mitre.org/data/definitions/119.html">CWE-119: Improper Restriction of Operations within the Bounds of a Memory Buffer</a></li>
```

Comment:
Why is this reference deleted?  Is it just that it will be automatically inserted due to the `cwe-119` tag in the query?

Reply:
Yes - the qhelp generator adds a basic CWE reference for all tags, so currently you get two references to the same CWE in the output for this query. For example: https://wiki.semmle.com/display/qhelptestcpp/Copy+function+using+source+size